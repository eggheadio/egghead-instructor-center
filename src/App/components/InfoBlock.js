import React from 'react'

import MoreInfoLink from './MoreInfoLink'

const InfoBlock = ({
  title,
  description,
  moreInfoUrl,
}) => (
  <div>
    <h3 className='f3'>
      {title}
    </h3>
    <div>
      {description}
      {moreInfoUrl
        ? <span className='ml2'>
            <MoreInfoLink url={moreInfoUrl} />
          </span>
        : null
      }
    </div>
  </div>
)

InfoBlock.propTypes = {
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string.isRequired,
  moreInfoUrl: React.PropTypes.string,
}

export default InfoBlock
