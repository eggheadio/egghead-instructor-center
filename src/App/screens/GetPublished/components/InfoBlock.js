import React from 'react'
import MoreInfo from './MoreInfo'

const InfoBlock = ({
  title,
  description,
  moreInfoUrl,
}) => (
  <div>
    <h3 className='f3'>
      {title}
    </h3>
    <div>
      {description}
      {moreInfoUrl
        ? <span className='ml2'>
            <MoreInfo url={moreInfoUrl} />
          </span>
        : null
      }
    </div>
  </div>
)

InfoBlock.propTypes = {
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string.isRequired,
  moreInfoUrl: React.PropTypes.string,
}

export default InfoBlock
