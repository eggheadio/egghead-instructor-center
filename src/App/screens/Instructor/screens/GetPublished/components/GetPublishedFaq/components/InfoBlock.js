import React, {PropTypes} from 'react'
import MoreInfo from '../../MoreInfo'

const InfoBlock = ({
  title,
  description,
  moreInfoUrl,
}) => (
  <div>
    <h3 className='f3'>
      {title}
    </h3>
    <div>
      {description}
      {moreInfoUrl
        ? <span className='ml2'>
            <MoreInfo url={moreInfoUrl} />
          </span>
        : null
      }
    </div>
  </div>
)

InfoBlock.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  moreInfoUrl: PropTypes.string,
}

export default InfoBlock
