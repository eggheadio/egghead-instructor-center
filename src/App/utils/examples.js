export default {
  text: {
    short: 'Lorem ipsum',
    medium: 'Lorem ipsum dolor sit amet, urna erat feugiat egestas, metus molestie, adipiscing sed nisl et, dictumst erat sem erat.',
  },
  url: 'https://egghead.io',
  markdown: 'Lorem ipsum *dolor*',
}
