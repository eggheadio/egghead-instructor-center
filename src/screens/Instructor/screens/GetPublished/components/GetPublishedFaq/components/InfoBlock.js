import React, {PropTypes} from 'react'
import Heading from 'components/Heading'
import MoreInfo from '../../MoreInfo'

const InfoBlock = ({
  title,
  description,
  moreInfoUrl,
}) => (
  <div>

    <Heading level='3'>
      {title}
    </Heading>

    <div>
      {description}
      {moreInfoUrl
        ? <span className='ml2'>
            <MoreInfo url={moreInfoUrl} />
          </span>
        : null
      }
    </div>

  </div>
)

InfoBlock.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  moreInfoUrl: PropTypes.string,
}

export default InfoBlock
