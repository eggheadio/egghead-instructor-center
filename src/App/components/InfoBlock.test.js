import React from 'react'
import renderer from 'react-test-renderer'

import examples from '../../App/utils/examples'

import InfoBlock from './InfoBlock'

test('Default', () => {
  const tree = renderer.create(<InfoBlock
    title={examples.text.short}
    description={examples.text.medium}
    moreInfoUrl={examples.url}
  />)
  expect(tree).toMatchSnapshot()
})
