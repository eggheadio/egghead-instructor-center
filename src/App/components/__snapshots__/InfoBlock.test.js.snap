exports[`test Default 1`] = `
<div>
  <h3
    className="f3">
    Lorem ipsum
  </h3>
  <div>
    Lorem ipsum dolor sit amet, urna erat feugiat egestas, metus molestie, adipiscing sed nisl et, dictumst erat sem erat.
    <span
      className="ml2">
      <a
        className="blue"
        href="https://egghead.io"
        target="_blank">
        <span
          className="fa fa-info-circle" />
      </a>
    </span>
  </div>
</div>
`;
